diff -ruN dxr3-0.2.7.orig/dxr3ffmpeg.c dxr3-0.2.7/dxr3ffmpeg.c
--- dxr3-0.2.7.orig/dxr3ffmpeg.c	2007-01-07 22:57:12.000000000 +0100
+++ dxr3-0.2.7/dxr3ffmpeg.c	2008-08-11 18:40:28.000000000 +0200
@@ -28,8 +28,10 @@
 cDxr3Ffmpeg::cDxr3Ffmpeg()
 {
     avcodec_init();
-    // Register only codec(s) we'll need.
-    register_avcodec(&mp2_decoder);
+    // Only the MP2 decoder would be needed, but individual registration of
+    // codecs bundled with ffmpeg itself does not appear to be really supported
+    // as of ffmpeg 2007-07-xx and later.
+    avcodec_register_all();
 }
 
 // ==================================
diff -ruN dxr3-0.2.7.orig/dxr3ffmpeg.h dxr3-0.2.7/dxr3ffmpeg.h
--- dxr3-0.2.7.orig/dxr3ffmpeg.h	2007-01-07 22:57:14.000000000 +0100
+++ dxr3-0.2.7/dxr3ffmpeg.h	2008-08-11 18:54:50.000000000 +0200
@@ -24,7 +24,7 @@
 
 extern "C"
 {
-#include <avcodec.h>
+#include <libavcodec/avcodec.h>
 }
 
 #include <string.h>
